test.js

```diff
@@ -34,6 +36,31 @@ test('.directory()', t => {
 	t.true(tempy.directory().includes(tmpdir()));
 });
 
+test('.write(string)', async t => {
+	const filePath = await tempy.write('unicorn', {name: 'test.png'});
+	t.is(fs.readFileSync(filePath, 'utf8'), 'unicorn');
+	t.is(path.basename(filePath), 'test.png');
+});
+
+test('.write(buffer)', async t => {
+	const filePath = await tempy.write(Buffer.from('unicorn'));
+	t.is(fs.readFileSync(filePath, 'utf8'), 'unicorn');
+});
+
+test('.write(stream)', async t => {
+	const readable = new stream.Readable({
+		read() { }
+	});
+	readable.push('unicorn');
+	readable.push(null);
+	const filePath = await tempy.write(readable);
+	t.is(fs.readFileSync(filePath, 'utf8'), 'unicorn');
+});
+
+test('.writeSync()', t => {
+	t.is(fs.readFileSync(tempy.writeSync('unicorn'), 'utf8'), 'unicorn');
+});
+
```

Comment:
There needs to b a lot more tests, especially regarding the stream handling, with tests for possible stream errors, incorrect streams being passed in, file system errors, etc. There's a lot that could go wrong with streams and we need to ensure we have good testing coverage.

Reply:
I've added tests for stream errors.